### PR TITLE
aid IDE hinting by overriding protected with more specialised interface

### PR DESCRIPTION
**Pull request by basz**
override properties with more specialised interface as annotations.

will make IDEs' happy (->save method)

### CODE REVIEW

**Comment by basz** [comment]
@prolic could you have a look pls, especially the second commit...

File: src/Service/AuthorizationCodeService.php [hunk 1]
```diff
@@ -33,6 +34,11 @@
 class AuthorizationCodeService extends AbstractTokenService
 {
     /**
+     * @var AuthorizationCodeRepositoryInterface
+     */
+    protected $tokenRepository;
```
Comment (prolic) [inline src/Service/AuthorizationCodeService.php]: Where is it set?

**Comment by basz** [comment]
https://github.com/zf-fr/zfr-oauth2-server/blob/master/src/Service/AbstractTokenService.php#L59

File: src/Service/AuthorizationCodeService.php (hunk 1 above)
Comment (prolic) [inline src/Service/AuthorizationCodeService.php]: When it is in abstract class, why add again?

**Comment by basz** [comment]
'cause in the extended classes it is known to be AuthorizationCodeRepositoryInterface which extends TokenRepositoryInterface

TokenRepositoryInterface doesn't specify ->save

**Comment by prolic** [comment]
I see it now. Becoming late here ;-)

**Pull request merged**